src/Pipes/GZip.hs

```diff
@@ -62,8 +62,8 @@ decompress' = Pipes.Zlib.decompress' gzWindowBits
 compress
   :: MonadIO m
   => Pipes.Zlib.CompressionLevel
-  -> Proxy x' x () B.ByteString m r -- ^ Decompressed stream
-  -> Proxy x' x () B.ByteString m r -- ^ Compressed stream
+  -> Producer B.ByteString m r -- ^ Decompressed stream
+  -> Producer' B.ByteString m r -- ^ Compressed stream
```

Comment:
Oh wait, I see what you mean.  Previously, for instance, it was possible to compress and decompress a `Pipe` (though maybe a bit weird?).

Well, I don't currently see how to restore that type of polymorphism to the `decompress` implementation.  (Any ideas?).  And I think that it makes a lot of sense for `compress` and `decompress` to have similar signatures.  And I also think that actually it's not unreasonable for these to be things that act on `Producer`s.

So I think I'm still happy with this pull request; but I do now understand that it's a non-trivial decision to accept it...